### PR TITLE
Process standards response from api

### PR DESCRIPTION
**Pull request by fareed-mohammed**

### CODE REVIEW

File: app/routes/v2/standards/index.js [hunk 1]
```diff
@@ -9,14 +10,36 @@ module.exports = [{
   options: {
     handler: async (request, h) => {
       const applyJourney = await cache.get('apply-journey', request.yar.id)
-      await sendStandardsRequestMessage(applyJourney, request.yar.id)
+      await sendStandardsRequestMessage({ sbi: applyJourney.selectedSbi.sbi, organistionId: applyJourney.selectedSbi.organisationId, callerId: applyJourney.callerId }, request.yar.id)
```
Comment (paulsimonandrews) [inline app/routes/v2/standards/index.js:13]: typo: `organistionId` -> `organisationId`

File: app/views/v2/add-standard-parcels/add-standard-parcels.njk [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          <div id="parcels-hint" class="govuk-hint">
+            You have {{ totalHa }}ha of eligible land for this standard
+          </div>
+
+          {% if checkboxItems.length > 0 %}
+            <div class="govuk-checkboxes">
+              {% for item in checkboxItems %}
+                <div class="govuk-checkboxes__item">
+                  <input class="govuk-checkboxes__input" id="parcels" name="parcels" type="checkbox" value="{{ item.value }}">
+                  <label class="govuk-label govuk-checkboxes__label" for="parcels">
```
Comment (paulsimonandrews) [inline app/views/v2/add-standard-parcels/add-standard-parcels.njk:37]: Set a unique `id` in the input and use that as the `for` in the label so you can select the label next to the checkbox. 

You could pass the id through from the ViewModel or use something like `id="parcels-{{loop.index}}"` and `for="parcels-{{loop.index}}"` directly in the nunjucks file

File: app/routes/v2/add-standards-parcels/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        }
+      )
+    }
+  },
+  {
+    method: 'POST',
+    path: '/v2/add-standard-parcels',
+    handler: async (request, h) => {
+      const applyJourney = await cache.get('apply-journey', request.yar.id)
+      const parcels = [...new Map(applyJourney.selectedStandard.parcels.map(item => [item.id, item])).values()]
```
Comment (paulsimonandrews) [inline app/routes/v2/add-standards-parcels/index.js:37]: I don't really understand what this is achieving. Why can't you just use:

`const parcels = applyJourney.selectedStandard.parcels`?

File: app/routes/v2/add-standards-parcels/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+      const selectedParcels = parcels.filter(item => request.payload.parcels.includes(item.id)).map(x => {
+        const object = {}
+
+        object.id = x.id
+        object.area = x.area
+        return object
+      })
```
Comment (paulsimonandrews) [inline app/routes/v2/add-standards-parcels/index.js:45]: Simplify the last map:

```
.map(x => ({
        id: x.id,
        area: x.area
      }))
```

File: app/routes/v2/add-standards-parcels/index.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+      const parcelArea = selectedParcels.reduce((accum, item) => accum + item.area, 0)
+      console.log(parcelArea)
```
Comment (paulsimonandrews) [inline app/routes/v2/add-standards-parcels/index.js:48]: Can probably remove this logging unless you find it useful

File: app/routes/v2/choose-level/index.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+},
+{
+  method: 'POST',
+  path: '/v2/choose-level',
+  handler: async (request, h) => {
+    const applyJourney = await cache.get('apply-journey', request.yar.id)
+
+    const level = request.payload.level
+    let selectedAmbitionLevel = null
+    selectedAmbitionLevel = applyJourney.paymentRates[level]
```
Comment (paulsimonandrews) [inline app/routes/v2/choose-level/index.js:29]: Just use a const?

`const selectedAmbitionLevel = applyJourney.paymentRates[level]`

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Looks good, working as expected with Option 1 backend.

A few comments/suggestions to simplify the code to consider.

Also, there isn't any validation on some of the routes, not sure if that was in scope here, but I think we should get this merged without the validation and add it later in another ticket.

File: app/views/v2/add-standard-parcels/add-standard-parcels.njk (hunk 2 above)
Comment (fareed-mohammed) [reply app/views/v2/add-standard-parcels/add-standard-parcels.njk:37]: Agreed

File: app/routes/v2/add-standards-parcels/index.js (hunk 3 above)
Comment (fareed-mohammed) [reply app/routes/v2/add-standards-parcels/index.js:37]: Well spotted, needs to be removed as duplicates issue fixed in the apply-api.

File: app/routes/v2/add-standards-parcels/index.js (hunk 4 above)
Comment (fareed-mohammed) [reply app/routes/v2/add-standards-parcels/index.js:45]: Agree

File: app/routes/v2/add-standards-parcels/index.js (hunk 5 above)
Comment (fareed-mohammed) [reply app/routes/v2/add-standards-parcels/index.js:48]: Removed

File: app/routes/v2/choose-level/index.js (hunk 6 above)
Comment (fareed-mohammed) [reply app/routes/v2/choose-level/index.js:29]: Agree

**Review by paulsimonandrews — APPROVED** [review]

**Pull request merged**